README.md

```diff
@@ -390,16 +425,44 @@ Example assertions:
       }
 ```
 
-## Convention based file tests
+### Injecting dependencies into tests
 
-Query tests can be written by providing the framework with a sql query file and a file with
+As an alternative to using `ThreadLocalTestContextHolder.testContext()` explicitly
+dependencies can be injected to test by framework. Field injection and argument injection
+through setup() method is supported.
+
+See example:
+```
+public class InjectionTest
```

Comment:
Added explanation. The meaning of MutableTablesState and ImmutableTablesState is explained before in the document